libbeat/common/converter/converter_test.go

```diff
@@ -0,0 +1,55 @@
+package converter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func TestBytes_Ntohs(t *testing.T) {
+
+	schema := Schema{
+		"lastname":    Str("name"),
+		"git":         Str("social.github"),
+		"number.nine": Int("nine"),
+	}
+
+	file, err := os.Open("example.json")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	defer file.Close()
+
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	var data common.MapStr
+	err = json.Unmarshal([]byte(byteValue), &data)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	conv(schema, data)
+}
```

Comment:
Could you also add a test that detects collisions like this one?
```
schema := Schema{
    "user": Str("name"),
    "user.git": Str("social.github"),
}
```
This should even panic as it should be considered an error in the code.